flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/AbstractFileSystemTable.java

```diff
@@ -53,15 +62,75 @@
 		context.getCatalogTable().getOptions().forEach(tableOptions::setString);
 		this.schema = TableSchemaUtils.getPhysicalSchema(context.getCatalogTable().getSchema());
 		this.partitionKeys = context.getCatalogTable().getPartitionKeys();
-		this.path = new Path(context.getCatalogTable().getOptions().getOrDefault(PATH.key(), PATH.defaultValue()));
-		this.defaultPartName = context.getCatalogTable().getOptions().getOrDefault(
-				PARTITION_DEFAULT_NAME.key(), PARTITION_DEFAULT_NAME.defaultValue());
+		this.path = new Path(tableOptions.get(PATH));
+		this.defaultPartName = tableOptions.get(PARTITION_DEFAULT_NAME);
 	}
 
-	static FileSystemFormatFactory createFormatFactory(ReadableConfig tableOptions) {
+	ReadableConfig formatOptions(String identifier) {
+		return new DelegatingConfiguration(tableOptions, identifier + ".");
+	}
+
+	FileSystemFormatFactory createFormatFactory() {
 		return FactoryUtil.discoverFactory(
 				Thread.currentThread().getContextClassLoader(),
 				FileSystemFormatFactory.class,
 				tableOptions.get(FactoryUtil.FORMAT));
 	}
+
+	@SuppressWarnings("rawtypes")
+	<F extends EncodingFormatFactory<?>> Optional<EncodingFormat> discoverOptionalEncodingFormat(
```

Comment:
Why not reuse `FactoryUtil.TableFactoryHelper#discoverOptionalEncodingFormat`?

Reply:
See below comments:
```
	/**
	 * Unlike {@link FactoryUtil#discoverFactory}, it will not throw an exception if it cannot
	 * find the factory.
	 */
```
Do you think we should modify `FactoryUtil.TableFactoryHelper#discoverOptionalEncodingFormat`?

Reply:
I will move these logic to `FileSystemTableFactory` and use `FactoryUtil` to create formats.